### ISSUE TITLE
Add X509Credential to JSON-LD context

### ISSUE THREAD

**Issue by reinkrul** [nuts-foundation/nuts-node#3600]

**Comment by stevenvegt** [comment]
I've created the following context, which can be played with on the [playground](https://tinyurl.com/24zrbb23).

Some things I discovered:
- We cannot use a colon `:` in the term name since it wants to expand it to a url, so `subject:L` is not usable. I opted for nesting the properties in a `https://nuts.nl/credentials#X509Subject`.
- For some reason the JSON-LD playground would not use me a type name starting with an X, so `X509Credential` did not work. Could be a fluke, bug or reserved keyword. Eventually I gave up and went for the `NutsX509Credential` type.
- Most values could be mapped on address properties from schema.org, but the `san` fields are weird. So I choose the oid code for the object type of the san extension and I created a new type `http://nuts.nl/credentials#SubjectAltName.otherName`. We could think of other solutions for this.

```json
{
  "@context": [
    "https://www.w3.org/2018/credentials/v1",
    {   
      "@version": 1.1,
      "@protected": true,
      "@propagate": true,

      "id": "@id",
      "type": "@type",
      
      "nuts": "https://nuts.nl/credentials#",
      "schema": "https://schema.org/",

      "NutsX509Credential": {
        "@id": "nuts:NutsX509Credential",

        "@context": {
          "@version": 1.1,
          "@protected": true,
          "@propagate": true,

          "subject": {
            "@id": "nuts:X509Subject",
            
            "@context": {
              "@version": 1.1,
              "@protected": true,
              "@propagate": false,

              "L":"schema:addressLocality",
              "C":"schema:addressCountry",
              "S": "schema:addressRegion",
              "O":"schema:name",
              "CN": "schema:url"
            }
          },
          "san": {
            "@id": "urn:oid:2.5.29.17",
            
            "@context": {
              "@version": 1.1,
              "@protected": true,
              "@propagate": false,

              "otherName":"nuts:SubjectAltName.othername"
            }
          }
        }
      }
    }
  ],
  "id": "did:web:nuts.nl:123#vc1",
  "type": ["VerifiableCredential", "NutsX509Credential"],
  
  "credentialSubject": {
    "id": "did:web:example.org:221",
    "subject": {
      "L": "'S-GRAVENHAGE",
      "C": "NL",
      "S": "Zuid-Holland",
      "O": "Tést Zorginstelling 01",
      "CN": "test.example.com"
      
    },
    "san": {
      "otherName":"2.16.528.1.1007.99.2110-1-900030787-S-90000380-00.000-11223344"
    }
  }
}
```
<img width="696" alt="Screenshot 2025-01-06 at 17 43 25" src="https://github.com/user-attachments/assets/810a58c7-9209-4d6b-842e-7d674d192f68" />

**Comment by reinkrul** [comment]
> We cannot use a colon : in the term name since it wants to expand it to a url, so subject:L is not usable. I opted for nesting the properties in a https://nuts.nl/credentials#X509Subject.

You could argue this is as intended, meaning we define a context for `subject`?

Parsing logic becomes a bit more complex then, though.

**Comment by stevenvegt** [comment]
> You could argue this is as intended, meaning we define a context for `subject`?
Yes, that is one option indeed. We can put the values directly into the credential subject, but that the "subject" and "san" values disappear. 

> Parsing logic becomes a bit more complex then, though.
Can you elaborate on that? More complex for whom?

**Comment by reinkrul** [comment]
`json-gold` also has the behavior with types starting with `X`, made an issue for it: https://github.com/piprate/json-gold/issues/81

We decided that, since we don't plan to actually support X509Credentials in JSON-LD format, to not have this issue block release of 6.1.0;
- we don't want parties to have to change their implementation again (we already renamed the VC once and changed its structure)
- we don't expect to run into issues, as we're planning to only issue JWT VCs in the future

**Comment by reinkrul** [comment]
Not doing this any more, since we're not processig it as JSON-LD anyways.
